Add postgres autodiscovery documentation (#19302)
DataDog/documentation: layouts/shortcodes/dbm-postgres-agent-config-examples.md

```

### Monitoring relation metrics for multiple logical databases

> In order to collect relation metrics (such as `postgresql.seq_scans`, `postgresql.dead_rows`, `postgresql.index_rows_read`, and `postgresql.table_size`), the Agent must be configured to connect to each logical database (by default, the Agent only connects to the `postgres` database). 

Reviewer: ```suggestion
In order to collect relation metrics (such as `postgresql.seq_scans`, `postgresql.dead_rows`, `postgresql.index_rows_read`, and `postgresql.table_size`), `database_autodiscovery` must be enabled because these metrics must be gathered across logical databases on the Postgres instance.
```

For version 7.48, specify a single "DBM" instance to collect DBM telemetry from all databases. 
In that instance, define a `database-autodiscovery` block. Set enabled to `True`. Then, specify the logical databases to monitor under the `include` list. Use a regular expression to match the names of the databases: for example, if `inventory` should be monitored, but not `inventory_1`, then specify `inventory$` in the include list. Databases can also be excluded from monitoring under `exclude`. 
If `include` is empty, the Agent will monitor all logical databases on the host. 

> The Agent will rediscover databases every 10 minutes by default. This parameter can be changed under the `refresh` configuration option.

Reviewer: ```suggestion
```

This is not necessary for a basic example

When using database autodiscovery, metrics on all tables will be collected for each database specified. For more granular relation monitoring per database, databases must be specified as distinct instances. This is described in [`Advanced Configuration`](/database_monitoring/setup_postgres/advanced_configuration#monitoring-relation-metrics-for-multiple-logical-databases). For Agent versions < 7.48, databases also must be specified as distinct instances.
```yaml
init_config:
instances:
  # This instance is the "DBM" instance. It will connect to the
  # `inventory` and `product` databases, and all databases prefixed by `user`
  # except for `users_deprecated`. It will send DBM telemetry from all databases,
  # but collect relation metrics on only the databases specified by `database_autodiscovery`.
  - dbm: true
    host: products-primary.123456789012.us-east-1.rds.amazonaws.com
    port: 5432
    username: datadog
    password: '<PASSWORD>'
    database_autodiscovery:

>       enabled: True

Reviewer: ```suggestion
      enabled: true
```

Should be yaml true, not python True

      include: 
        - inventory$
        - products$
        - user*
      exclude:
        - users_deprecated$

>       refresh:
>         - 1000

Reviewer: ```suggestion
```

I think this should be removed. The default value is fine here.

>     dbstrict: true

Reviewer: ```suggestion
```
`dbstrict` is no longer needed here

    relations:
      - relation_regex: .*
        relkind:
          - r

>           - i

Reviewer: Can you remove relkind?
